### PR TITLE
#762 - Create and Delete Tasks

### LINKED ISSUE #762: Add/Delete Task Functionality
**We have a feature branch `feature/#448-task-lists` for this epic. Make sure to branch from/PR into this branch!**

Must be completed after #760 #761 Get Tasks Endpoint, Create Task Endpoint, and Delete Task Endpoint (replace with ticket references once made).

Make datagrids populate with the actual task data from the backend. Hook up the delete button from the action button on the right. When clicked it should pop up a modal that asks if you're sure you want to delete the task. If the user clicks 'confirm' the task should be deleted, if they click 'cancel', it should go back to the state before the modal was opened. 

Wire up the add task button as shown below. When the add task button is clicked, a new row should appear in the data grid, allowing you to type in what you want for each of the fields, and having save and delete buttons on the left. Assignee should be an [multiple value autocomplete textfield](https://mui.com/material-ui/react-autocomplete/#combo-box)(multiple values) with options of only members of the project’s team, tasks will be able to have multiple assignees The delete button should just cancel the new task and take you back to before you pressed add task, the save button should actually create the new task and add it to the current data grid. 

image

### PR DESCRIPTION
**Pull request by RChandler234**
## Changes

Added in add new task flow.

## Notes

Went ahead and got started on this one since we want to release this Wednesday, but this is still a draft PR. Please feel free to leave review, suggestions, or do some of it yourself cause we need to get this one done asap.

Got most of the ui stuff done today, still need to get some of the edit modes done (particularly the multiselect autocomplete for the assignees, and some sort of datepicker for the deadline)

Also the delete task button isn't hooked up, just waiting on the backend PR to be done for that.

## Screenshots

image

## To Do

- [ ] Get datepicker working for deadline
- [ ] Get multiselect working for assignees
- [ ] finish few details of create hookup
- [ ] hookup delete endpoint once that's done

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #762

### CODE REVIEW

File: src/frontend/src/apis/tasks.api.ts [hunk 1]
```diff
@@ -3,9 +3,16 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
+import { WbsNumber, wbsPipe } from 'shared';
 import axios from '../utils/axios';
 import { apiUrls } from '../utils/urls';
 
+export const createSingleTask = (wbsNum: WbsNumber, payload: any) => {
```
Comment (anthonybernardi) [inline src/frontend/src/apis/tasks.api.ts:10]: we want to move away from these any payload blocks and pass in the actual parameters

Comment (anthonybernardi) [inline src/frontend/src/apis/tasks.api.ts:10]: also documentation

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx [hunk 2]
```diff
@@ -118,7 +118,7 @@ const ProjectViewContainer: React.FC<ProjectViewContainerProps> = ({ proj, enter
         actionButton={projectActionsDropdown}
       />
       <ProjectDetails project={proj} />
-      <TaskList tasks={proj.tasks} />
+      <TaskList tasks={proj.tasks} currentProject={proj.wbsNum} />
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx:121]: currentProjectWbsNumber

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
delete endpoint is done so you can do that part now too

one thing is that if the page only shows 5 tasks, and there are 5 tasks and you press create, it starts the creation thingy on the next page without taking you to the next page

also i can type in the actions cell


i think i would just prefer a modal lmao

**Comment by RChandler234** [comment]
merged in peyton's branch, also the change in editworkpackage is because transforming the date that way causes a bug where it can set the date to the next day (since it was using utc time not our current time)

**Comment by RChandler234** [comment]
moved the create task to the top of the task list so you don't have to click onto the next page to see it

**Review by anthonybernardi — APPROVED** [review]
500 line file is gross but code looks solid, if we fix that error im good with this

**Pull request merged**